src/test/java/com/microsoft/azure/relay/HybridConnectionListenerTest.java

```diff
@@ -71,6 +75,49 @@ public void openAndCloseTest() {
 		assertEquals("Listener offline handler was not called exactly once", 1, offlineHandlerCalled.get());
 	}
 	
+	@Test
+	public void customHeadersTest() {
+		String key = "GoodHeader";
+		String value = "GoodValue";
+		Map<String, List<String>> customHeaders = new HashMap<String, List<String>>();
+		customHeaders.put(key, Arrays.asList(new String[]{value}));
+		
+		CompletableFuture<Void> requestReceived = new CompletableFuture<Void>();
```

Comment:
>requestReceived [](start = 26, length = 15)

Doesn't someone need to wait on this in case an exception was thrown?

Reply:
Ooops, deleted the wait afterwards by accident, will add it.

---
In reply to: [266164217](https://github.com/Azure/azure-relay-java/pull/35#discussion_r266164217) [](ancestors = 266164217)